src/pyats/contrib/creators/netbox.py

```diff
@@ -212,8 +227,8 @@ def _generate(self):
         headers = { "Authorization": token }
         data = {}
         topology = {}
-        devices_url = self._format_url(self._netbox_url, 
-                                                "api/dcim/devices/?format=json")
+        url=f"api/dcim/devices/?format=json&{self._url_filter}"
```

Comment:
if self._url_filter is '' it would introduce `&` at the end of url.

is that intended? if not should probably if/else it

Reply:
Agreed on self._url_filter, the extra & at the end won't hurt anything, but it should not be there and if/else makes sense.

The PR does not modify the "interface not present" logic.  I proposed wrapping it around with a self._topology flag, since current implementation does not work as its not providing "link" attribute which appears to be required for topology section.  I just added a todo flag to fix that section.

Reply:
link is not required in topology, you can define interfaces without links

Reply:
I will review this and provide details of the issues I was seeing with topology.

I have added another commit with a couple more changes/additions.  I am trying to get this to a "usable" state in our netbox deployment and running into a lot of interesting caveats.

Reply:
I have isolated the "bug" with topology, and updated the PR, issue was due to missing interface type, not link (my apologies).  Let me know your thoughts on this.

Reply:
I would like to propose keeping the topology build flag, as for large netbox environment looping though each interface/IP takes a considerable amount of time for a large inventory/testbed.  

Again, I am very new to PyATS and not quite grasping the importance of topology yet, but in our implementation of Netbox, we are not quite 100% with documenting all of the topology/links properly to get meaningful data out of it.